lib/queue.rb

```diff
@@ -1,31 +1,60 @@
 class Queue
 
   def initialize
-    # @store = ...
-    raise NotImplementedError, "Not yet implemented"
+    @queue = Array.new(20)
+    @capacity = @queue.count
+    @front = 0
+    @back = 0
+  end
+
+  def full
+    @front == @back && self.size == @capacity
   end
 
   def enqueue(element)
-    raise NotImplementedError, "Not yet implemented"
+    raise QueueFullError if self.full
+    @queue[@back] = element
+    @back = (@back + 1) % @queue.length
   end
 
   def dequeue
-    raise NotImplementedError, "Not yet implemented"
+    return nil if self.empty?
+    dequeued = @queue[@front]
+    @queue[@front] = nil
+    @front = (@front + 1) % @queue.length
+    return dequeued
   end
 
   def front
-    raise NotImplementedError, "Not yet implemented"
+    return nil if @queue.empty?
+    return @queue[@front]
   end
 
   def size
```

Comment:
Why not just use the positions of front and back to calculate the size instead of this O(n) operation.